src/RenderJSONVisitor.cc

```diff
@@ -0,0 +1,136 @@
+//
+//  RenderJSONVisitor.cc
+//  libdrafter
+//
+//  Created by Pavan Kumar Sunkara on 17/06/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#include "RenderJSONVisitor.h"
+#include "sosJSON.h"
+#include "stream.h"
+
+using namespace refract;
+
+namespace drafter
+{
+
+    RenderJSONVisitor::RenderJSONVisitor(const sos::Base::Type& type_)
+    : type(type_) {}
+
+    RenderJSONVisitor::RenderJSONVisitor()
+    : type(sos::Base::UndefinedType) {}
+
+    void RenderJSONVisitor::assign(sos::Base value) {
+        if (type == sos::Base::ArrayType) {
+            pArr.push(value);
+        }
+        else if (type == sos::Base::UndefinedType) {
+            result = value;
+        }
+    }
+
+    void RenderJSONVisitor::assign(std::string key, sos::Base value) {
+        if (!key.empty() && type == sos::Base::ObjectType) {
+            pObj.set(key, value);
+        }
+    }
+
+    void RenderJSONVisitor::visit(const IElement& e) {
+        if (e.element() == "object") {
```

Comment:
No, there is no other way how to do it.

In general visitor pattern is "wrong name" Better definitio of visitor is
> Visitor lets you define a new operation without changing the classes of the elements on which it operates.

see paragraph " Understanding what problem Visitor addresses" there it explain how it work http://www.artima.com/cppsource/top_cpp_aha_moments.html